patches/tModLoader/Terraria/ModLoader/Engine/ErrorReporting.cs

```diff
@@ -1,141 +1,142 @@
+			tip = Language.GetTextValue("tModLoader.ModExceptionHint");
+		else if (e is IOException && e.Message.Contains("cloud file provider")) {
+			if (string.IsNullOrEmpty(e.HelpLink))
+				e.HelpLink = "https://github.com/tModLoader/tModLoader/wiki/Basic-tModLoader-Usage-FAQ#save-data-file-issues";
+			tip = Language.GetTextValue("tModLoader.OneDriveHint");
+			if (Language.ActiveCulture == null) // This error typically happens before localization is loaded, so fallback to english text.
+				tip = "Tip: Try installing/enabling OneDrive. Right click your Documents folder and enable \"Always save on this device\"";
+		}
+		else if (e is SynchronizationLockException)
+			tip = Language.GetTextValue("tModLoader.AntivirusHint");
+		else if (e is TypeInitializationException)
+			tip = Language.GetTextValue("tModLoader.TypeInitializationHint");
+
+		if (e.HelpLink != null) {
+			try {
+				Utils.OpenToURL(e.HelpLink);
+			}
+			catch { }
+		}
+
+		if (tip != null)
+			message += "\n\n" + tip;
+
+		message += "\n\n" + e;
+
+		FatalExit(message);
+	}
+
+	/// <summary>
+	/// Shows an OS-provided modal message box displaying a message and a number of buttons and returns the button index of the user-selected option. The first option will be mapped to return key and the last option to escape key. The options are displayed from right to left in order.
+	/// </summary>
+	/// <returns></returns>
+	internal static int ShowMessageBoxWithChoices(string title, string message, string[] buttonLabels)
+	{
+		SDL2.SDL.SDL_MessageBoxButtonData[] buttons = new SDL2.SDL.SDL_MessageBoxButtonData[buttonLabels.Length];
+		for (int i = 0; i < buttonLabels.Length; i++) {
+			buttons[i] = new SDL2.SDL.SDL_MessageBoxButtonData() { flags = 0, buttonid = i, text = buttonLabels[i] };
+			if (i == 0)
+				buttons[i].flags = SDL2.SDL.SDL_MessageBoxButtonFlags.SDL_MESSAGEBOX_BUTTON_RETURNKEY_DEFAULT;
+			else if (i == buttonLabels.Length - 1)
+				buttons[i].flags = SDL2.SDL.SDL_MessageBoxButtonFlags.SDL_MESSAGEBOX_BUTTON_ESCAPEKEY_DEFAULT;
+		}
+		SDL2.SDL.SDL_MessageBoxData messageBoxData = new SDL2.SDL.SDL_MessageBoxData() {
+			flags = SDL2.SDL.SDL_MessageBoxFlags.SDL_MESSAGEBOX_INFORMATION,
+			window = IntPtr.Zero,
+			title = title,
+			message = message,
+			numbuttons = buttons.Length,
+			buttons = buttons,
+			colorScheme = null
+		};
+		int buttonID;
+		if (SDL2.SDL.SDL_ShowMessageBox(ref messageBoxData, out buttonID) < 0) {
+			Logging.tML.Info("ShowMessageBoxWithChoices: Error displaying message box");
+		}
+		if (buttonID == -1) {
+			Logging.tML.Info("ShowMessageBoxWithChoices: No selection");
+		}
+		return buttonID;
+	}
+
+	/// <summary> Various error codes to show in Visual Studio. Mainly used to cross reference with source code. Subject to change if more granular error codes are needed. </summary>
+	internal enum TMLErrorCode
+	{
+		TML001, // Other exception
+		TML002, // Any BuildException
+		TML003, // tMod file in use
+	}

```

Comment:
It seems like you've accidentally changed file line endings, since this whole file is showing as changed